manifests/init.pp

```diff
@@ -157,16 +157,11 @@
     require => Exec['RHSM-register'],
   }
 
-  if $repo_extras {
-    ::rhsm::repo { "rhel-${::operatingsystemmajrelease}-server-extras-rpms":
-      require => Exec['RHSM-register'],
+  unless(empty($enabled_repo_ids)) {
+    $enabled_repo_ids.each | $repo_id | {
+      ::rhsm::repo { $repo_id:
```

Comment:
please do not add the leading `::`, they are not needed anymore since we do not support Puppet3.